tests/unittests/test_chunk_size.py

```diff
@@ -0,0 +1,96 @@
+        ['positive_int_chunk_size', {'chunk_size': 5}, [5, 2]],
+        ['positive_int_string_chunk_size', {'chunk_size': '5'}, [5, 2]],
+        ['empty_string_chunk_size', {'chunk_size': ''}, [100, 1]],
+    ])
+    def test_chunk_size_valid(self, name, test_data, expected_data):
+        '''
+            Test case to verify we set the valid integer as the chunk_size
+        '''
+
+        sorted_campaigns = [1, 2, 3, 4, 5, 6, 7, 8]
+
+        client = MailchimpClient({**self.config, **test_data})
+        reports_email_activities = ReportEmailActivity({}, client, {**self.config, **test_data}, Catalog(), [], [])
+        chunk_campaigns = reports_email_activities.chunk_campaigns(sorted_campaigns, 0)
+
+        # Verify the chunk_size value
+        self.assertEqual(reports_email_activities.chunk_size, expected_data[0])
+        # Verify the list of campaign chunks
+        self.assertEqual(len(list(chunk_campaigns)), expected_data[1])
+
+    @parameterized.expand([
+        ['zero_int_chunk_size', {'chunk_size': 0}, [True, 'The chunk_size cannot be Zero(0).']],
+        ['zero_string_chunk_size', {'chunk_size': '0'}, [True, 'The chunk_size cannot be Zero(0).']],
+        ['negative_int_chunk_size', {'chunk_size': -5}, [True, 'The chunk_size cannot be negative.']],
+        ['negative_int_string_chunk_size', {'chunk_size': '-5'}, [True, 'The chunk_size cannot be negative.']],
+        ['positive_float_chunk_size', {'chunk_size': 5.1}, [False]],
+        ['positive_float_string_chunk_size', {'chunk_size': '5.1'}, [False]],
+        ['string_chunk_size', {'chunk_size': 'test'}, [False]],
+        ['negative_float_chunk_size', {'chunk_size': -5.1}, [False]],
+        ['negative_float_string_chunk_size', {'chunk_size': '-5.1'}, [False]],
+        ['positive_float_near_zero_chunk_size', {'chunk_size': 0.1}, [False]],
+        ['positive_float_string_near_zero_chunk_size', {'chunk_size': '0.1'}, [False]],
+        ['negative_float_near_zero_chunk_size', {'chunk_size': -0.1}, [False]],
+        ['negative_float_string_near_zero_chunk_size', {'chunk_size': '-0.1'}, [False]],
+    ])
+    @mock.patch("tap_mailchimp.streams.LOGGER.info")
+    def test_chunk_size_error(self, name, test_data, expected_data, mocked_logger_info):
+        '''
+            Test cases to verify we raise an error if invalid chunk_size is passed from the config
+        '''
+
+        client = MailchimpClient({**self.config, **test_data})
+
+        # Verify we raise ValueError
+        with self.assertRaises(ValueError) as e:
+            ReportEmailActivity({}, client, {**self.config, **test_data}, Catalog(), [], [])
+
+        # Verify the error message
+        self.assertEqual(str(e.exception), 'Please provide a valid integer value for the chunk_size parameter.')
+
+        # Verify we get the logger when Negative and Zero(0) chunk_size is passed in the config
+        if expected_data[0]:
+            mocked_logger_info.assert_called_with(expected_data[1])
```

Comment:
Keep scenario-specific test suites in two different test cases rather than using the if condition.

Reply:
Updated the test case to separate tests.